substra/sdk/schemas.py

```diff
@@ -183,6 +183,45 @@ class _BaseComputePlanSpec(_Spec, abc.ABC):
     aggregatetuples: Optional[List[ComputePlanAggregatetupleSpec]]
     testtuples: Optional[List[ComputePlanTesttupleSpec]]
 
+    def get_dependency_graph(self):
+        """Get the tuple dependency graph and, for each type of tuple, a mapping table id/tuple.
+        """
+        tuple_graph = dict()
+        traintuples = dict()
+        if self.traintuples:
+            for traintuple in self.traintuples:
+                if traintuple.in_models_ids is not None:
+                    tuple_graph[traintuple.traintuple_id] = traintuple.in_models_ids
```

Comment:
What do you think of adding a check making sure that all IDs are unique? It could be checking that traintuple.traintuple_id is not already in tuple_graph for example. This could prevent some really weird behavior.

Reply:
yes, checking that the user did not give two tuples with the same id makes sense ! I think it is checked server side too but I don't know what kind of behaviour it might induce in the auto batching 👍